File: Week_09/java/src/_415_AddStrings.java
/**
 * 第一遍：2020/08/04周二 ✅
 * 第二遍：2020/08/06周四 ✅
 * 第三遍：2020/08/07周五 ✅
 * 第三遍：2020/07/08周四
 */
class _415_AddStrings {
    public String addStrings(String num1, String num2) {
        int i = num1.length() - 1;
        int j = num2.length() - 1;
        StringBuilder ret = new StringBuilder();
        int add = 0;
        while (i >= 0 || j >= 0 || add != 0) {
            int a = i >= 0 ? num1.charAt(i) - '0' : 0;
            int b = j >= 0 ? num2.charAt(j) - '0' : 0;
            int s = a + b + add;
            ret.append(s % 10);
            add = s / 10;
            i--;
            j--;
        }
        return ret.reverse().toString();
    }
}

File: Week_09/kotlin/src/_415_AddStrings.kt
fun addStrings(num1: String, num2: String): String {
    var i = num1.length - 1
    var j = num2.length - 1
    val ret = StringBuilder()
    var add = 0
    while (i >= 0 || j >= 0 || add != 0) {
        val a = if (i >= 0) num1[i] - '0' else 0
        val b = if (j >= 0) num2[j] - '0' else 0
        val s = a + b + add
        ret.append(s % 10)
        add = s / 10
        i--
        j--
    }
    return ret.reverse().toString()
}